packages/priority-queue/src/index.js

```diff
@@ -96,7 +96,11 @@ export const createQueue = () => {
 			const callback = /** @type {WPPriorityQueueCallback} */ ( elementsMap.get(
 				nextElement
 			) );
-			callback();
+
+			if ( typeof callback === 'function' ) {
```

Comment:
another option would be a try/catch here instead, with a warning on catch so at least there is an indication that something is going through the queue without a callback

Reply:
in which case the callback won't be a valid function?

Reply:
@youknowriad It turns out that in some cases if the callBack in runWaitingList takes a while to complete, the element can be added back to the waitList before `elementsMap.delete` is called. It then gets called before the new element in the waitingList goes through runWaitingList, so when it does there is no matching element in the elementsMap and the exception happens.

<img width="525" alt="Screen Shot 2021-02-10 at 2 53 34 PM" src="https://user-images.githubusercontent.com/3629020/107459225-943c4900-6bba-11eb-9ad2-4c11669ab36f.png">

I added some extra debugging in this PR, and you can see above callback being called for waitingList item 82, but then item 83 being added before the delete from element map happens for 82.

I have changed this PR to handle this situation by check that an element has not been added back to the waitList before deleting it from the element map, and this seems to fix the issue in the gallery refactor PR, but not sure if there are any other implications to handling this way.